### PR TITLE
all: use bakery.v2

### PR DESCRIPTION
**Pull request by rogpeppe**
As this is an API-breaking change, this becomes v4.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/370/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/371/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmrepo-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/charmrepo-merge/104

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/373/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/374/

File: csclient/csclient_test.go [hunk 1]
```diff
@@ -1,10 +1,11 @@
 // Copyright 2015 Canonical Ltd.
 // Licensed under the LGPLv3, see LICENCE file for details.
 
-package csclient_test // import "gopkg.in/juju/charmrepo.v3/csclient"
+package csclient_test // import "gopkg.in/juju/charmrepo.v4/csclient"
 
 import (
 	"bytes"
+	"context"
```
Comment (mhilton) [inline csclient/csclient_test.go]: I'm not certain if we should use the built in context or not here. Is it used anywhere else in the project?

Comment (rogpeppe) [reply csclient/csclient_test.go]: I think it's OK - juju can use the stdlib context, and this is probably sufficiently in that realm.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/376/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmrepo-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/charmrepo-merge/105

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/377/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**